documentation/design_proposal/001_organization.md

```diff
@@ -0,0 +1,223 @@
+- Rename of a organization is doable
+- Easier to use since it's clear that a team/user is under a specific organiaztion.
+
+### Cons
+
+- Needs to be applied after the deployment is done.
+- API-url needs to be provided to the CR together with username & secret.
+
+### CRD
+
+```.yaml
+apiVersion: integreatly.org/v1alpha1
+kind: Organization
+metadata:
+  name: organization-example
+spec:
+  - name: organiaztionX
+      grafana:
+        - url: http://grafana/
+          username: admin # default value
+          password:
+            secretName: mysecret
+              key: password
+        - url: https://external-grafana/
+          username: admin # default value
+          password:
+            secretName: admin-secret2
+              key: password
+          tls:
+            secretName: external-https-cert
+      users:
+        - login: user1
+          email: user1@github.com
+          role: admin
+          name: user1
+          theme: light
+          isGrafanaAdmin: true
+          isDisabled: true
+          password:
+            secretName: usersecret
+            key: password
+        - login: user2
+          email: noreply@grafana.com
+          role: Viewer
+      teams:
+        - name: MyTestTeam
+          email: email@test.com
+          theme: dark
+          homeDashboardId: 39
+          timezone: utc
+          memebers:
+            - user1
+            - user2
+```
+
+### Options
+
+The organaiztion CRD could also use a label selector just like we do for dashboardSelectors but the other way around.
+Instead of the grafana instance finding which organizations to add the organaization will find which grafana instances to apply to.
+
+This way we woulden't need to define organiaztion.spec.grafana.
+
+## Context
+
+We need to consider the [multi-namespace support](https://github.com/grafana-operator/grafana-operator/pull/599) that is currently getting worked on.
+The operator will start to manage multiple grafana instances in a single cluster.
+
+In the future it might be intresting to [manage external grafana instances](https://github.com/grafana-operator/grafana-operator/issues/402) even though it's currently not being prioratized.
+We haven't had any discussion about this but that might also be off the table considering [Avoiding scope creep](https://github.com/grafana-operator/grafana-operator/issues/497).
+
+When creating the new controller we should use [CreateOrUpdate](https://github.com/grafana-operator/grafana-operator/issues/362) from the begining.
+
+## Alternatives
+
+Instead of having a single CRD to manage organiaztions, teams and users we could have multiple ones.
+
+If teams, users or orgs get more "abilities" they are separated and can be changed independently.
+I just feel it might assist in organizing all parts of the user/right management.
+
+Also if upstream grafana decides to do a breaking change around one the CR:s we only need to bump one of the API versions,
```

Comment:
This is very true, given my last point as well. I think we need to think of how to balance this, Although we don't really support any other version of upstream grafana (other than the one we define in our constants), it is still a valid use case for a lot of users, since they do opt to use a newer version of grafana despite the warning that we provide.
In the end, this is really only an issue for us if we decide to upgrade the version of grafana that the operator ships with.
I'll make a note to reach out to legal, to see if we can upgrade the grafana version to the one with the new licence